File: ThePalindrome.java
public class ThePalindrome {
  public int find(String s) {
    String tail = "";
    int i = 0;

    while (!isReversible(s + tail)) {
      tail = (s.charAt(i) + "") + tail;
      i++;
    }

    return (s + tail).length();
  }

  public boolean isReversible(String s) {
    StringBuilder sb = new StringBuilder(s);
    return sb.reverse().toString().equals(s);
  }
}

File: the_palindrome.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
#include <functional>
using namespace std;

class ThePalindrome
{
public:
  int find(string s)
  {
    int n = s.size();
    for (int i = 0; i < n; i++)
    {
      string ss = s.substr(i);
      if (is_reversible(ss))
      {
        return i + n;
      }
    }
    return -1;
  }

private:
  bool is_reversible(string s)
  {
    int n = s.size();
    for (int i = 0; i < n / 2; i++)
    {
      if (s[i] != s[n - i - 1])
        return false;
    }
    return true;
  }
};
